### PR TITLE
fix: replace deprecated package `io/ioutil` with `os` or `io`

### PR DESCRIPTION
**Pull request by ulbqb**
## Description

Replace deprecated `io/ioutil` package since Go 1.16.

* ioutil.TempDir -> os.MkdirTemp
* ioutil.WriteFile -> os.WriteFile
* ioutil.ReadAll -> io.ReadAll
* ioutil.ReadFile -> os.ReadFile

### CODE REVIEW

**Review by Mdaiki0730 — APPROVED** [review]

File: p2p/key.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func LoadOrGenNodeKey(filePath string) (*NodeKey, error) {
 
 // LoadNodeKey loads NodeKey located in filePath.
 func LoadNodeKey(filePath string) (*NodeKey, error) {
-	jsonBytes, err := ioutil.ReadFile(filePath)
+	jsonBytes, err := os.ReadFile(filePath)
```
Comment (tnasu) [inline p2p/key.go:73]: nit: can use ostracon/libs/os/os.go: ReadFile

File: privval/file.go [hunk 2]
```diff
@@ -206,7 +206,7 @@ func LoadFilePVEmptyState(keyFilePath, stateFilePath string) *FilePV {
 
 // If loadState is true, we load from the stateFilePath. Otherwise, we use an empty LastSignState.
 func loadFilePV(keyFilePath, stateFilePath string, loadState bool) *FilePV {
-	keyJSONBytes, err := ioutil.ReadFile(keyFilePath)
+	keyJSONBytes, err := os.ReadFile(keyFilePath)
```
Comment (tnasu) [inline privval/file.go:209]: nit: can use ostracon/libs/os/os.go: ReadFile

File: privval/file.go [hunk 3]
```diff
@@ -224,7 +224,7 @@ func loadFilePV(keyFilePath, stateFilePath string, loadState bool) *FilePV {
 	pvState := FilePVLastSignState{}
 
 	if loadState {
-		stateJSONBytes, err := ioutil.ReadFile(stateFilePath)
+		stateJSONBytes, err := os.ReadFile(stateFilePath)
```
Comment (tnasu) [inline privval/file.go:227]: nit: can use ostracon/libs/os/os.go: ReadFile

File: state/state.go [hunk 4]
```diff
@@ -340,7 +340,7 @@ func MakeGenesisStateFromFile(genDocFile string) (State, error) {
 
 // MakeGenesisDocFromFile reads and unmarshals genesis doc from the given file.
 func MakeGenesisDocFromFile(genDocFile string) (*types.GenesisDoc, error) {
-	genDocJSON, err := ioutil.ReadFile(genDocFile)
+	genDocJSON, err := os.ReadFile(genDocFile)
```
Comment (tnasu) [inline state/state.go:343]: nit: can use ostracon/libs/os/os.go: ReadFile

File: statesync/chunks.go [hunk 5]
```diff
@@ -209,7 +208,7 @@ func (q *chunkQueue) load(index uint32) (*chunk, error) {
 	if !ok {
 		return nil, nil
 	}
-	body, err := ioutil.ReadFile(path)
+	body, err := os.ReadFile(path)
```
Comment (tnasu) [inline statesync/chunks.go:211]: nit: can use ostracon/libs/os/os.go: ReadFile

File: types/genesis.go [hunk 6]
```diff
@@ -154,7 +154,7 @@ func GenesisDocFromJSON(jsonBlob []byte) (*GenesisDoc, error) {
 
 // GenesisDocFromFile reads JSON data from a file and unmarshalls it into a GenesisDoc.
 func GenesisDocFromFile(genDocFile string) (*GenesisDoc, error) {
-	jsonBlob, err := ioutil.ReadFile(genDocFile)
+	jsonBlob, err := os.ReadFile(genDocFile)
```
Comment (tnasu) [inline types/genesis.go:157]: nit: can use ostracon/libs/os/os.go: ReadFile

**Review by tnasu — COMMENTED** [review]
Thanks for your changes. I left the upstream PR. It's a far future work for us.
* https://github.com/tendermint/tendermint/pull/9206

**Review by tnasu — APPROVED** [review]

**Review by torao — APPROVED** [review]

**Pull request merged**